File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    const n = height.length;

    let maxarea = 0;

    let low = 0;
    let high = n - 1;

    while (low <= high) {
        let area = (high - low) * (Math.min(height[low], height[high]));
        maxarea = Math.max(area, maxarea);
        if (height[low] <= height[high]) {
            low++;
        }
        else {
            high--;
        }

    }


    return maxarea;

};

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int low = 0;
        int high = n - 1;
        int maxarea = 0;
        while (low < high) {
            maxarea =
                max(maxarea, ((high - low) * min(height[low], height[high])));
            if (height[low] < height[high]) {

                low++;

            } else {
                high--;
            }
        }
        return maxarea;
    }
};